core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```diff
@@ -104,7 +104,10 @@ object ConsoleConsumer extends Logging {
     Runtime.getRuntime.addShutdownHook(new Thread() {
       override def run() {
         consumer.stop()
-
+        // if we generated a random group id (as none specified explicitly) then avoid polluting zookeeper with persistent group data, this is a hack
+        if (!conf.groupIdPassed && conf.options.has(conf.zkConnectOpt))
+          ZkUtils.maybeDeletePath(conf.options.valueOf(conf.zkConnectOpt), "/consumers/" + conf.consumerProps.get("group.id"))
```

Comment:
Yes, although AdminUtils already offers 'deleteConsumerGroupInZK' and 'deleteConsumerGroupInfoForTopicInZK' to implement this, I notice that ConsoleConsumer originally employ this snippet of code to delete inactive console consumer group.
The key point here is we must add this kind of code in addShutdownHook otherwise this thread might not delete the unused zk nodes if the first auto-commit task did not get started.